### PR TITLE
Feature/add config for underscore switch

### PR DESCRIPTION
**Pull request by lbriais**
Maybe you can have a look at this quickie:

A new config flag is introduced `friendly_symbols` which defaults to `true` that provides the standard behaviour (ie '-' becomes '_'). If set to `false` leaves option symbol untouched (see test introduced)...

Maybe you can have a look, because I didn't feel very proud to have to duplicate this config entry into two classes: `Option` and `Options`.

Regards,

### CODE REVIEW

**Comment by leejarvis** [comment]
Thanks @lbriais.

I actually wouldn't bother with an additional method. I appreciate the composition considerations, but I think the code is so simple and as you mention, it introduces a bit of a hack in using `respond_to?`, so I think duplicate is warranted. I also wouldn't bother with a separate method for the same reason. I'm happy to add a few lines of code.

I think your config name is fine, but perhaps something more explicit like `underscore_flags` might be nicer. WDYT?

This way `Option#key` might look something like this:

```ruby
key = config[:key] || flags.last.sub(/\A--?/, '')
key = key.tr("-", "_") if underscore_flags?
key
```

**Comment by lbriais** [comment]
Agreed for the removal of the extra module with its method, but my duplication concern was more about the `friendly_symbols` (or `underscore_flags` if you prefer) as it has to be duplicated in both `Option::CONFIG` and `Options::CONFIG`...

**Comment by leejarvis** [comment]
> as it has to be duplicated in both Option::CONFIG and Options::CONFIG

I'm not sure it does. The [global config is passed into the option specific config](https://github.com/leejarvis/slop/blob/master/lib/slop/options.rb#L52), so whatever is default in `Options::CONFIG` will be the default in the per option config. So we could actually omit it from the option specific config hash. That said, since we'll be checking it elsewhere in the `Option` code, it probably makes sense to have an option specific default too. I think this duplication is OK, and really it only has a bearing on anyone using the `Option` API directly, and I suspect that's probably not many people.

**Comment by lbriais** [comment]
Please check PR modified. 

BTW I confirm that if you remove the option from `Option::CONFIG`, then test _'converts dashes to underscores to make multi-word options symbol-friendly'_ doesn't pass...

File: lib/slop/option.rb [hunk 1]
```diff
@@ -1,8 +1,10 @@
 module Slop
   class Option
+
```
Comment (leejarvis) [inline lib/slop/option.rb]: Superfluous whitespace here

File: lib/slop/result.rb [hunk 2]
```diff
@@ -7,6 +7,7 @@ module Slop
   # result of those actions. This class contains the important most used
   # methods.
   class Result
+
```
Comment (leejarvis) [inline lib/slop/result.rb]: Superfluous whitespace here

File: test/option_test.rb [hunk 3]
```diff
@@ -21,6 +21,12 @@ def option(*args)
       assert_equal :foo_bar, option(%w(-f --foo-bar), nil).key
     end
 
+    it "when specified, it won't convert dashes to underscores to make multi-word options symbol-friendly" do
+      assert_equal :'foo-bar', option(%w(-f --foo-bar), nil, underscore_flags: false).key
+    end
+
+
+
```
Comment (leejarvis) [inline test/option_test.rb]: Superfluous whitespace here

**Comment by leejarvis** [comment]
> BTW I confirm that if you remove the option from Option::CONFIG, then test 'converts dashes to underscores to make multi-word options symbol-friendly' doesn't pass...

Right, yep. That's because the test uses the `Option` API directly. I think your changes are good though, great work!

I added some comments about whitespace changes, but otherwise happy to merge. Feel free to mention the config option in the README if you think it's worth it.

Thanks again! 💜

**Comment by lbriais** [comment]
I applied the requested patches for superfluous lines introduced.
I will let you deal with the option description as the whole documentation about `CONFIG` is actually missing.

Thank you for your support, and eager to see that released upstream and updated gem published.

Regards

- Laurent -

**Comment by lbriais** [comment]
Humm I'm afraid I diverted from the PR review process... Sorry

**Comment by leejarvis** [comment]
@lbriais Thanks. For your information I pulled down your branch and squashed your commits by running: `git rebase -i master` and then squashing and rewriting the commit message: 70e1de88583b934e583aad2b7c2bbffda4a4ea57

I'll merge this to master and update the CHANGELOG. There's some further adjustments I plan to make before cutting a new release, but I suspect it'll be done soon.

Thanks again for your contribution! 💜 💚

**Pull request closed without merging**